src/gluonts/model/naive_2/_predictor.py

```diff
@@ -133,30 +134,21 @@ class Naive2Predictor(RepresentablePredictor):
         Length of the seasonality pattern of the input data
     """
 
-    @validated()
-    def __init__(
-        self,
-        prediction_length: int,
-        freq: Optional[str] = None,
-        season_length: Optional[int] = None,
-    ) -> None:
-        super().__init__(prediction_length=prediction_length)
+    prediction_length: int
+    freq: Optional[str] = None
+    season_length: Optional[int] = None
 
+    def __post_init_post_parse__(self):
         assert (
-            season_length is None or season_length > 0
+            self.season_length is None or self.season_length > 0
         ), "The value of `season_length` should be > 0"
-        assert freq is not None or season_length is not None, (
+        assert self.freq is not None or self.season_length is not None, (
             "Either the frequency or season length of the time series "
             "has to be specified. "
         )
 
-        self.freq = freq
-        self.prediction_length = prediction_length
-        self.season_length = (
-            season_length
-            if season_length is not None
-            else get_seasonality(freq)
-        )
+        if self.season_length is None:
+            self.season_length = get_seasonality(self.freq)
```

Comment:
This we should be able to simplify using the `OrElse` approach I've added.

The idea is that you just specify this:

```py
season_length: int = serde.OrElse(lambda freq: get_seasonality(self.freq))
```

If one passes a value we take that, "or else" we use `get_seasonality`.